### PR TITLE
Filter by category

### PR DESCRIPTION
**Pull request by i-zor066**
This is where I'm currently at. Everything works.

### CODE REVIEW

File: BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/adapter/TabAdapter.java [hunk 1]
```diff
@@ -32,15 +33,45 @@ public TabAdapter(FragmentManager fm, CharSequence Titles[], int NumbOfTabsumb)
...
             return pointsOfInterestMapFragment;
         }
 
     }
 
+    public void updateCategory(String category) {
+        try {
+            pointsOfInterestFragmentList.updateCategory(category);
+            pointsOfInterestMapFragment.updateCategory(category);
+        } catch (NullPointerException e) {
```
Comment (Electryc) [inline BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/adapter/TabAdapter.java]: `NullPointerException`s should never be caught as they hinder a problem in the code. Why could the fragments be null here? Do you call `updateCategory()` before they're created? 

As a last resort if you want to guard against the null you can do something like this at the beginning of the method (if both of the fragments are null at some point): 

``` java
if (pointsOfInterestFragmentList == null || pointsOfInterestMapFragment == null) {
    return;
}
```

The above will stop the method executing if any of the fragment is null. Is there any case in which one of them is null and the other not? In this case the guard could be more granular, as in: 

``` java
if (pointsOfInterestFragmentList!=null) {
  pointsOfInterestFragmentList.updateCategory(category);  
}

if (pointsOfInterestMapFragment!=null) {
   pointsOfInterestMapFragment.updateCategory(category); 
}
```

The same goes for the rest of the `try catch` for `NullPointerException`. In any case, catching this exception is very unadvisable.

File: BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/fragment/PointsOfInterestMapFragment.java [hunk 2]
```diff
@@ -45,28 +51,46 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+        } else {
+            pointsOfInterest = BlocSpotApplication.getSharedDataSource().getPointsOfInterestForCategory(currentCategory);
+        }
+
+        int size = pointsOfInterest.size();
 
         float lat = pointsOfInterest.get(0).getLatitude();
         float lon = pointsOfInterest.get(0).getLongitude();
 
+        googleMap.clear();
```
Comment (Electryc) [inline BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/fragment/PointsOfInterestMapFragment.java:84]: :+1:

**Comment by Electryc** [comment]
Apart from the `NullPointerException` `try catch` , it's looking good :+1:

**Comment by i-zor066** [comment]
The NullPointerExceptions were actually only appearing if rotation was changed at any time, since I noticed that rotation also causes NPEs with the dialog fragments I just locked the whole thing to portrait only.

**Comment by Electryc** [comment]
Normally, locking the whole app in portrait should be a product decision, not driven by development ease let's say :smiley_cat: Let's merge this now as it is, and then maybe we can take a look later what's the problem in there :+1:

**Pull request merged**